optax/_src/loss.py

```diff
@@ -358,3 +358,33 @@ def loop_body(prev, x):
   per_seq_loss = -jnp.einsum('bn,bn->b', logalpha_phi_last, one_hot)
 
   return per_seq_loss
+
+
+def kl_div_loss(predictions: chex.Array,
+                targets: chex.Array,
+                log_targets: bool = False) -> chex.Array:
+  """Computes the Kullback-Leibler divergence (relative entropy) loss.
+
+  Measures the information gain achieved if target probability distribution
+  would be used instead of predicted probability distribution.
+
+  References:
+    [Kullback, Leibler, 1951](https://www.jstor.org/stable/2236703)
+
+  Args:
+    predictions: Probabilities of predicted distribution with shape [..., dim].
+    Expected to be in the log-space to avoid underflow.
+    targets: Probabilities of target distribution with shape [..., dim].
+    Expected to be strictly positive.
+    log_targets: Specifies whether targets are given in the log-space.
+
+  Returns:
+    Kullback-Leibler divergence of predicted distribution from target
+    distribution with shape [...].
+  """
+  chex.assert_type([predictions, targets], float)
+  if log_targets:
+    loss = jnp.multiply(jnp.exp(targets), jnp.subtract(targets, predictions))
```

Comment:
I know this is a matter of style and I originally called this a _nit_, but we have since decided that we would like to enforce the `+, -, *, /`-style for optax. For large parts of optax, the expressions are too complex mathematically to be readable with `jnp.multiply` etc in our opinion and we'd like to keep the style of mathematical expressions consistent across the codebase.

Could you change your function to use the `*`, `-` operators? Thanks a lot!